### PR TITLE
tsbua: Partial load interface

### PR DESCRIPTION
**Pull request by tylershunt**
Interface to load part of an RDG

After #182  (or maybe I should open this as a pull request for Gurbinder's branch not sure).

I imitate partial parquet load (which @scober is working on) with slicing; (but i commented out slicing because the CuSP code currently expects a fully populated table.

Fixes point 2 of #218

### CODE REVIEW

**Comment by tylershunt** [comment]
Again depends on @gurbinder533's  code so only the last commit will be in the final PR.

**Comment by gurbinder533** [comment]
I am working on landing #182. CI builds for GPU are failing at the moment as ignored GPU path for now.

**Comment by tylershunt** [comment]
> Where do you think we should check if the property exists in the schema or not so that we can give a better error message?

We could add a more specific error code? Something like "PropertyNotFound"? It doesn't seem that weird to support that since we're treating properties as first class citizens anyway.

**Comment by gurbinder533** [comment]
Yup, that sounds good.

**Comment by tylershunt** [comment]
> We could add a more specific error code? Something like "PropertyNotFound"? It doesn't seem that weird to support that since we're treating properties as first class citizens anyway.

Added in #235

File: libcusp/include/galois/graphs/NewGeneric.h [hunk 1]
```diff
@@ -206,32 +207,37 @@ class NewDistGraphGeneric : public DistGraph<NodeTy, EdgeDataTy> {
...
 
-    Result<std::shared_ptr<OutIndexView>> g_result =
-        galois::OutIndexView::Make(filename, node_properties, edge_properties);
+    auto open_res =
+        tsuba::Open(filename, tsuba::kReadOnly | tsuba::kReadPartial);
+    if (!open_res) {
+      GALOIS_LOG_FATAL("failed to open (filename {}) OutIndexView: {}",
+                       filename, open_res.error());
+    }
+    tsuba::RDGHandle handle = std::move(open_res.value());
```
Comment (witchel) [inline libcusp/include/galois/graphs/NewGeneric.h:216]: I'm kind of a std::move n00b, but shouldn't this work without std::move?  Is this the move constructor or move assignment?  I know, this is where I include a gotbolt link that settles the whole issue....

File: libtsuba/include/tsuba/RDG.h [hunk 2]
```diff
@@ -70,9 +70,30 @@ struct RDG {
...
+  uint64_t version;
+  uint64_t edge_type_size;
+  uint64_t num_nodes;
+  uint64_t num_edges;
+};
+
+/// includes the header and the list of indexes
+struct GRPrefix {
+  tsuba::GRHeader header;
+  uint64_t out_indexes[]; /* NOLINT length is defined by num_nodes_ in header */
```
Comment (witchel) [inline libtsuba/include/tsuba/RDG.h:83]: Wow, this C idiom survives?  I thought there would be some C++ variadic wizardry that obviates it.

Comment (tylershunt) [reply libtsuba/include/tsuba/RDG.h:83]: It's a struct that maps onto memory that `FileView` is managing. If someone knows a cleaner more C++ way to get a variable length, structured view of raw data I'm all ears.

File: libcusp/include/galois/graphs/NewGeneric.h (hunk 1 above)
Comment (tylershunt) [reply libcusp/include/galois/graphs/NewGeneric.h:216]: Yeah you'r right, handle is trivially copy-able so I can simplify. I just got in the habit of moving out of `galois::Result<..>`

Comment (tylershunt) [reply libcusp/include/galois/graphs/NewGeneric.h]: Actually an `RDGFile` wrapper is better here since it handle's the close.

**Comment by tylershunt** [comment]
I'm going to hold off on any merges until I get enthusiastic approval from @gurbinder533 since this touches a lot of what he's been up to.

**Comment by tylershunt** [comment]
> It looks GOOOOD. I tested it locally as well.
> Thank you :).

BUt wait are you sure?

File: libtsuba/include/tsuba/RDG.h (hunk 2 above)
Comment (tylershunt) [reply libtsuba/include/tsuba/RDG.h:83]: Oh I guess I could build an ArrowArray? I'll leave that for the future.

**Pull request merged**